### DISCUSSION TITLE
Proper modeling technique for racks with "cable management" and slot covers

### DISCUSSION THREAD

**Discussion by gustojoel** [netbox-community/netbox#11978] [Q&A]
greetings everyone!

In our rack plans, we have cable managers, spacers, and covers that are "installed" in all of the rack units. This should be pretty standard to keep hot/cold aisles working properly, and for marking space that is deliberately held open for heat dissipation between devices.

With what I know about how to handle this currently, this results in a lot of extra "devices", as each one of these plastic "things" in a rack U is its own _unique_ device.

**Q:** I am trying to understand if there is a better way: if there is a way to have a single "device" that exists in multiple places simultaneously? It's obviously for a pretty specific use-case, but I think lots of people would find it useful as it's basically for anything that doesn't have a serial number/isn't electronic.

**Additional Justification:** The reason this is so important is to properly and accurately track "rack space available". Do you  consider your rack "full" when it's got 42U worth of spacers in it? Perhaps you do, and perhaps that is room for expansion. The calculations for "rack space available" would need some additional logic to respect things like:
- cable managers (they are there for a reason)
- empty units that are there for cooling between two devices (not available for more equipment)
- finally, a whole bunch of rack units that are indeed _available_ - but have covers on them for proper heat management.

Thank you for taking the time to read, and process what I am thinking about at the moment.
== Joel

**Comment by joaolucasmacedo** [comment]
> cable managers (they are there for a reason)
You can model these as devices or as rack reservations.

>     empty units that are there for cooling between two devices (not available for more equipment)
The same way, you can create rack reservations.

>     finally, a whole bunch of rack units that are indeed available - but have covers on them for proper heat management.

Rack covers can be devices, or rack reservations.

**Reply by bluikko** [reply]
> Rack covers can be devices, or rack reservations.

If you do this then all racks will be 100% full... I believe that was the point of the OP.

**Comment by candlerb** [comment]
> finally, a whole bunch of rack units that are indeed available - but have covers on them for proper heat management.

I would simply not model those at all.  This is free space, which has to be blocked off for no reason other than airflow, and when you install a device you throw them away.  I see no reason for these to be in Netbox: it just has to be policy that unused space is blanked out appropriately.

If you leave them out of modelling, then everything just works with regards free space.  As you say, cable management and spacing between devices for cooling *should* be modelled, as this is not "free" space.
